legendKing.py

```diff
@@ -181,36 +216,146 @@ def setup_scale(vpName, maxValue, minValue, guide, reverse=None,
+
+    # elif (reverse == True) and colormap != 'Symmetric': #pull a quick swap there
+    #     color1, color2 = color2, color1
+    
+    
+    "STEP 3: Create interpolated list of hex colors between bounding colors"
+    if colormap == 'Symmetric':
+        bot_colors = generate_interpolated_hex_table(color1_ext,color1_mid,N)
+        top_colors = generate_interpolated_hex_table(color2_mid,color2_ext,N)
+        bot_colors.append('Grey80') #central pivot point should be grey
+        bot_colors.extend( top_colors ) #merge the two lists
+        
+        hex_colors = bot_colors
+    elif colormap in ['Cbs-cool', 'Cbs-warm']:   #self-created colormap tables
+        hex_colors = generate_interpolated_hex_table(color1, color2,N)
+    elif colormap in ['Viridis','Plasma']:
+        print('ARE WER HERE')
+        # Python 2.7 implimentation
+        mpl_cmap = cm.get_cmap(colormap.lower(), N).colors
+        if reverse:  mpl_cmap = np.flip(mpl_cmap, axis=0)
+        # Python 3.5 + implimentation (untested -- waiting for Abaqus v2024 support of python 3+)
+        #mpl_cmap = matplotlib.colormaps[colormap.lower()].resample(N).colors
+        
+        hex_colors = convert_mpl_colortable( mpl_cmap )
+        
+    "STEP 3: Create the session spectrum object with name associated with user choice earlier"
+    if colormap in ['Symmetric','Cbs-cool','Cbs-warm','Viridis','Plasma']:
+        abaqus.session.Spectrum( name=colormap, colors=hex_colors)
+
+def convert_mpl_colortable(mpl_cmap):
+    "Matplotlib returns table of RBGA values. This converts the table to a list of hex colors"
+    hex_colors = ['#{:02x}{:02x}{:02x}'.format(
+                int(255 * c[0] ),
+                int(255 * c[1] ),
+                int(255 * c[2]))  for c in mpl_cmap]
+    return hex_colors
+        
+
+# (TS) 
+def generate_interpolated_hex_table(color1, color2, N):
+    """
+    Pass this function 2 colors and it'll linearly ramp HSV values between the two
+    >> No guarantee that this is isochromatic
+    """
+    import colorsys
+    import numpy as np
+    Hvals = np.linspace(color1[0],color2[0],N)
+    Svals = np.linspace(color1[1],color2[1],N)
+    Vvals = np.linspace(color1[2],color2[2],N)
+    
+    HSV_table = np.ones((N,3))*np.nan
+    HSV_table[:,0] = Hvals
+    HSV_table[:,1] = Svals
+    HSV_table[:,2] = Vvals
+    
+    hex_colors = ['#{:02x}{:02x}{:02x}'.format(
+            int(255 * colorsys.hsv_to_rgb(h/360, s/100, v/100)[0]),
+            int(255 * colorsys.hsv_to_rgb(h/360, s/100, v/100)[1]),
+            int(255 * colorsys.hsv_to_rgb(h/360, s/100, v/100)[2]))
+            for h, s, v in HSV_table]   #ungodly list comprehension
+    return list(hex_colors)
 
 
 def readSettings():
     "Read settings file or create a new settings if necessary"
     if settings:
         return  # Already read
     try:
-        with open(jsonFileName) as f:
+        with open(jsonFileName, encoding='utf8') as f:
```

Comment:
Unfortunately, the current versions of Abaqus CAE are still using Python2.7 which does not support an `encoding` argument in `open()`. We should remove this from both lines for now.